src/workerd/io/actor-sqlite.c++

```
@@ -195,6 +197,21 @@ void ActorSqlite::ExplicitTxn::rollbackImpl() noexcept(false) {
   }
 }
 
+void ActorSqlite::onCriticalError(int sqliteErrorCode, kj::StringPtr message) {
+  switch (sqliteErrorCode) {
+    case SQLITE_FULL:
+      JSG_FAIL_REQUIRE(Error, "Database or disk full: ", message);
+    case SQLITE_IOERR:
+      JSG_FAIL_REQUIRE(Error, "Disk I/O error: ", message);
+    case SQLITE_BUSY:
+      JSG_FAIL_REQUIRE(Error, "Database in use by another process: ", message);
+    case SQLITE_NOMEM:
+      JSG_FAIL_REQUIRE(Error, "Out of memory: ", message);
+    case SQLITE_INTERRUPT:
+      JSG_FAIL_REQUIRE(Error, "Out of memory: ", message);
+  }
+}
```

Probably good to put a KJ_UNREACHABLE down here to make sure that the code never silently falls off the end of the function.

---

This code has changed, this is no longer needed - https://github.com/cloudflare/workerd/pull/3728/commits/567cf308fdd05af33e3265118445c24452ff84e9